web/main.go

```diff
@@ -180,7 +218,7 @@ func (s *server) authUser(r *http.Request) (user *pb.User, errFunc httpErrorWrit
 		return nil, badRequest, errors.Wrap(err, "failed to decode cookie")
 	}
 
-	userResp, err := s.getUser(userID)
+	userResp, err := s.getUser(trace.NewContext(ctx, span.NewChild("retrieve_user")), userID)
```

Comment:
This is a bit unexpected/unintuitive for me. I don't quite understand why the called method should be aware of the caller and the fact that caller must have passed a parent ctx that it should be deriving a new span from?